components/NavDrawer/NavDrawer.test.tsx

```diff
@@ -0,0 +1,182 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import {
+  Drawer,
+  DrawerContent,
+  DrawerDescription,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from './NavDrawer';
+
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(),
+    removeListener: jest.fn(),
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
+  })),
+});
+
+beforeAll(() => {
+  window.scrollTo = jest.fn();
+});
+
+describe('DrawerHeader Component', () => {
+  it('renders with default classes', () => {
+    render(<DrawerHeader data-testid="drawer-header" />);
+
+    const header = screen.getByTestId('drawer-header');
+    expect(header).toHaveClass(
+      'grid h-16 items-center gap-1.5 px-0 text-center sm:text-left',
+    );
+  });
+});
+
+describe('DrawerFooter Component', () => {
+  it('renders with default classes', () => {
+    render(<DrawerFooter data-testid="drawer-footer" />);
+
+    const footer = screen.getByTestId('drawer-footer');
+    expect(footer).toHaveClass('mt-auto flex flex-col gap-2 p-4');
+  });
+});
+
+describe('DrawerTitle Component', () => {
+  it('renders with default classes', () => {
+    render(
+      <Drawer>
+        <DrawerTitle data-testid="drawer-title" />
+      </Drawer>,
+    );
+
+    const title = screen.getByTestId('drawer-title');
+    expect(title).toHaveClass(
+      'mr-auto text-lg font-semibold leading-none tracking-tight',
+    );
+  });
+
+  it('renders the correct title', () => {
+    render(
+      <Drawer>
+        <DrawerTitle data-testid="drawer-title">Drawer Title</DrawerTitle>
+      </Drawer>,
+    );
+
+    const title = screen.getByTestId('drawer-title');
+    expect(title).toHaveTextContent('Drawer Title');
+  });
+
+  it('renders the close button with X icon', () => {
+    render(
+      <Drawer>
+        <DrawerTitle data-testid="drawer-title" />
+      </Drawer>,
+    );
+
+    const closeButton = screen.getByTestId('drawer-close');
+    expect(closeButton).toBeInTheDocument();
+    expect(closeButton.querySelector('svg')).toBeInTheDocument();
```

Comment:
could this better reflect the test's intention? "renders the close button with X icon"
how do we know from line 84 that the svg is an X icon?